vms/platformvm/vm.go

```diff
@@ -270,6 +271,31 @@ func (vm *VM) Initialize(
 	return nil
 }
 
+func (vm *VM) PruneMempool() {
+	removedTxs := set.Set[*txs.Tx]{}
+	droppedTxs := make(map[ids.ID]error)
+
+	vm.Builder.Iterate(func(tx *txs.Tx) bool {
+		_, _, dropped, err := vm.Builder.VerifyTxAgainstAcceptedState(tx)
+		if err != nil {
+			if dropped {
+				droppedTxs[tx.ID()] = err
+			}
+
+			removedTxs.Add(tx)
+		}
+
+		// Always return true to iterate through all the txs in the mempool.
+		return true
+	})
```

Comment:
I don't think we're going to actually want to verify the txs in the iteration function. I think we should get all the txs out of the mempool then verify them one-by-one.